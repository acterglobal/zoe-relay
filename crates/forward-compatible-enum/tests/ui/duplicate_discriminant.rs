use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
pub enum DuplicateDiscriminant {
    #[discriminant(1)]
    Variant1,
    #[discriminant(1)]  // Duplicate!
    Variant2,
    Unknown { discriminant: u32, data: Vec<u8> },
}