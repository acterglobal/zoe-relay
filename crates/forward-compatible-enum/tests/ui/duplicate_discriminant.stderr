error: duplicate discriminant value: 1
 --> tests/ui/duplicate_discriminant.rs:7:5
  |
7 | /     #[discriminant(1)]  // Duplicate!
8 | |     Variant2,
  | |____________^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/duplicate_discriminant.rs:10:2
   |
10 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/duplicate_discriminant.rs`
